pkg/publicshare/publicshare.go

```diff
@@ -213,8 +219,8 @@ func IsCreatedByUser(share link.PublicShare, user *user.User) bool {
 }
 
 // IsWriteable checks if the grant for a publicshare allows writes or uploads.
-func IsWriteable(g *link.Grant) bool {
-	p := g.Permissions.Permissions
-	return p.CreateContainer || p.Delete || p.InitiateFileUpload ||
+func IsWriteable(perm *link.PublicSharePermissions) bool {
+	p := perm.GetPermissions()
+	return p != nil && p.CreateContainer || p.Delete || p.InitiateFileUpload ||
 		p.Move || p.AddGrant || p.PurgeRecycle || p.RestoreFileVersion || p.RestoreRecycleItem
```

Comment:
Wouldn't this still panic when `p == nil`? It will evaluate `p != nil && p.Createcontainer` (which is false), then continue with evaluating `p.Delete` (which panics). Or do I miss some magic here?

Reply:
I think you're right. I'm missing parentheses there but let me check this quickly to confirm.

Reply:
Fixed it.